service/bf_suggestion.go

```diff
@@ -0,0 +1,216 @@
+package service
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"strconv"
+	"strings"
+
+	"github.com/evergreen-ci/evergreen"
+	"github.com/evergreen-ci/evergreen/model/task"
+	"github.com/evergreen-ci/evergreen/util"
+	"github.com/mongodb/grip"
+	"github.com/pkg/errors"
+)
+
+func getBFSuggestionClient(bbProj evergreen.BuildBaronProject) *bfSuggestionClient {
+	if bbProj.BFSuggestionServer == "" {
+		return nil
+	}
+	return &bfSuggestionClient{
+		bbProj.BFSuggestionServer,
+		bbProj.BFSuggestionUsername,
+		bbProj.BFSuggestionPassword}
+}
+
+const suggestionPath = "/suggestions/{task_id}/{execution}"
+const getFeedbackPath = "/feedback/{task_id}/{execution}/user_id/{user_id}"
+const sendFeedbackPath = "/feedback/{task_id}/{execution}"
+const removeFeedbackPath = "/feedback/{task_id}/{execution}/user_id/{user_id}/feedback_type/{feedback_type}"
+
+type bfSuggestionClient struct {
+	server   string
+	username string
+	password string
+}
+
+// Send a request to the BF Suggestion server.
+func (bfsc *bfSuggestionClient) request(ctx context.Context, req *http.Request) ([]byte, error) {
+	client := util.GetHTTPClient()
+	defer util.PutHTTPClient(client)
+
+	if bfsc.username != "" {
+		req.SetBasicAuth(bfsc.username, bfsc.password)
+	}
+
+	req = req.WithContext(ctx)
+	resp, err := client.Do(req)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return nil, errors.Wrapf(err, "Failed to read HTTP response body (status: %s)", resp.Status)
+	}
+
+	if resp.StatusCode >= 300 || resp.StatusCode < 200 {
```

Comment:
Might you actually just want to check `!= 200`? You should also use the constant `http.StatusOK` from the standard library.